src/app/shared/filter-by-property/filterbyproperty.pipe.ts

```diff
@@ -0,0 +1,31 @@
+import {Pipe, PipeTransform} from '@angular/core';
+
+@Pipe({
+    name: 'filterByProperty',
+})
+export class FilterbypropertyPipe implements PipeTransform {
+    transform<T, P extends keyof T>(
+        items: T[] | undefined | null,
+        searchingProperty: P,
+        searchValue: T[P],
+    ): T[] | undefined | null {
+        if (!items) {
+            return items;
+        }
+
+        if (typeof searchValue === 'string') {
+            return items.filter(item => {
+                const itemProperty = (item[searchingProperty] as string).toLowerCase();
+                const value = (searchValue as string).toLowerCase();
+
+                if (itemProperty.includes(value)) {
+                    return item;
+                }
+
+                return null;
```

Comment:
Рекомендация:
Мы из `filter` метода массива должны вернуть `boolean`, по этому данный кусок кода можно было бы свести к:
```suggestion
                return itemProperty.includes(value);
```